integreat_compass/cms/views/offers/offer_delete_view.py

```diff
@@ -0,0 +1,40 @@
+import logging
+
+from django.contrib import messages
+from django.contrib.auth.mixins import PermissionRequiredMixin
+from django.core.exceptions import PermissionDenied
+from django.urls import reverse_lazy
+from django.utils.translation import gettext as _
+from django.views.generic.edit import DeleteView
+
+from ...models import Offer
+
+logger = logging.getLogger(__name__)
+
+
+class OfferDeleteView(PermissionRequiredMixin, DeleteView):
+    """
+    View showing all of a user's offers
+    """
+
+    model = Offer
+    http_method_names = ["post"]
+    success_url = reverse_lazy("cms:protected:my_offers")
+
+    def get_permission_required(self):
+        offer_instance = Offer.objects.filter(pk=self.request.POST.get("pk")).first()
+        if offer_instance.creator != self.request.user:
+            raise PermissionDenied()
+        return ["cms.change_offer"]
+
+    def form_valid(self, form, *args, **kwargs):
+        title = (
+            Offer.objects.filter(pk=self.request.POST.get("pk"))
+            .first()
+            .latest_version.title
+        )
+        messages.success(
+            self.request,
+            _('The offer "{}" has successfully been deleted.').format(title),
```

Comment:
```suggestion
        messages.success(
            self.request,
            _('The offer "{}" has successfully been deleted.').format(
                self.object.latest_version.title
            ),
```
No need for an additional query, the base Django view already retrieves the offer from the database.

Apart from that: If you don't perform any additional checks, it's preferred to use `get()` in the first place instead of `filter().first()`, since if no such object exists, the error `Offer matching query does not exist` is more informative than `'NoneType' object has no attribute 'latest_version'`.

Reply:
Alright, thanks for letting me know!!